executor/simpletest/simple_test.go

```diff
@@ -712,6 +712,29 @@ func TestUser(t *testing.T) {
 	dropUserSQL = `DROP USER IF EXISTS 'test1'@'localhost' ;`
 	tk.MustExec(dropUserSQL)
 
+	// Test create/alter user with `tidb_auth_token`
+	tk.MustExec(`CREATE USER token_user IDENTIFIED WITH 'tidb_auth_token' REQUIRE token_issuer 'issuer-abc'`)
+	tk.MustQuery(`SELECT plugin, token_issuer FROM mysql.user WHERE user = 'token_user'`).Check(testkit.Rows("tidb_auth_token issuer-abc"))
+	tk.MustExec(`ALTER USER token_user REQUIRE token_issuer 'issuer-123'`)
+	tk.MustQuery(`SELECT plugin, token_issuer FROM mysql.user WHERE user = 'token_user'`).Check(testkit.Rows("tidb_auth_token issuer-123"))
+	tk.MustExec(`ALTER USER token_user IDENTIFIED WITH 'tidb_auth_token'`)
+	tk.MustExec(`CREATE USER token_user1 IDENTIFIED WITH 'tidb_auth_token'`)
+	tk.MustQuery(`show warnings`).Check(testkit.RowsWithSep("|", "Warning|1105|TOKEN_ISSUER is needed for 'tidb_auth_token' user, please use 'alter user' to declare it"))
```

Comment:
> - When creating a new user with tidb_auth_token plugin, if no attribute or no email is declared, TiDB server would NOT give any warning or error, since the attribute could be modified by ALTER USER statements. 

This is the original specification of this feature, could you double confirm if this is expected?(Although the behavior is different, IMHO this is fine).

Reply:
IMHO, I think it is expected. The warning is just for the missing `token_issuer`, not the `email` attribute, though they are both needed in the later authentication. 

To be honest I think all the warning for this auth plugin is not necessary for our general users. If they indeed miss something in `create user`, `alter user` can repair. The warning is just a small reminder.